### PR TITLE
Chant Detail: refactor and tweak concordances summary

### PR DESCRIPTION
**Pull request by jacobdgm**
This PR primarily refactors the concordances summary displayed on the Chant Detail page, creating a new function in order to decrease nesting and adding type annotations.

It also makes a couple of tweaks to this summary:
- the Slovak Early Music Database is now called Cantus Planus in Slovakia, 
- the databases in the summary are now sorted by the number of concordant chants they contain. 

This PR works towards solving #1033. It was initially meant to add two new databases to the list of databases, but these databases no longer appear in the concordances provided by Cantus Index, so this feature is currently neither testable nor necessary.

Screenshots to illustrate sorting and updated spelling:
Before:
Screenshot 2023-09-29 at 1 42 25 PM
After (much neater!):
Screenshot 2023-09-29 at 1 42 33 PM

### CODE REVIEW

File: django/cantusdb_project/main_app/views/chant.py [hunk 1]
```diff
@@ -67,6 +67,139 @@
...
+        database_summary: dict = {
+            "name": name,
+            "initialism": initialism,
+            "base_url": base_url,
+            "results_url": results_url,
+            "results_count": results_count,
+        }
+        database_summaries.append(database_summary)
+
+    filtered_database_summaries: List[dict] = [
```
Comment (dchiller) [inline django/cantusdb_project/main_app/views/chant.py]: Maybe clearer to put this conditional before you append to `database_summaries` in the `for` block above.

File: django/cantusdb_project/main_app/views/chant.py [hunk 2]
```diff
@@ -67,6 +67,139 @@
...
+    )
+
+    gregorien_response: Optional[Response] = None
+    try:
+        gregorien_response = requests.get(
+            f"https://gregorien.info/chant/cid/{cantus_id}/en",
+            timeout=5,
+        )
+    except (SSLError, Timeout) as exc:
+        print(  # eventually, we could log this rather than printing it
```
Comment (dchiller) [inline django/cantusdb_project/main_app/views/chant.py:181]: Where does this get printed to? The docker logs?

**Review by dchiller — APPROVED** [review]

File: django/cantusdb_project/main_app/views/chant.py (hunk 2 above)
Comment (jacobdgm) [reply django/cantusdb_project/main_app/views/chant.py:181]: I don't know! Let me check!

Comment (jacobdgm) [reply django/cantusdb_project/main_app/views/chant.py:181]: Yes, they do end up in the docker logs for our django container.

**Pull request merged**